yamux/src/connection/stream.rs

```diff
@@ -131,6 +132,14 @@ impl Stream {
         matches!(self.shared().state(), State::Closed)
     }
 
+    pub fn is_acknowledged(&self) -> bool {
+        self.shared().acknowledged
+    }
```

Comment:
Why expose this to the user? For the test above only? If so, can we move the test instead?
```suggestion

    pub(crate) fn is_acknowledged(&self) -> bool {
        self.shared().acknowledged
    }
```

Reply:
I think in the context of `rust-yamux`, this is useful information. Yes, it also allows us to keep the test next to all the others.

I am not too worried about this being in the public API because it is encapsulated within `libp2p-yamux` anyway. A nice benefit is that our tests are written against a more stable interface instead of reaching into internal data structures.

Reply:
> I think in the context of `rust-yamux`, this is useful information

In what scenario would a user use it?

Reply:
Until the point where the stream is not acknowledged, this is essentially equivalent to asking a QUIC connection whether it is 0RTT. Anything you write to a stream before it is acknowledged might be discarded by the remote.